packages/block/src/header.ts

```diff
@@ -491,9 +489,12 @@ export class BlockHeader {
     return this._common.hardfork() || this._common.activeHardfork(this.number.toNumber())
   }
 
-  private async _getBlockByHash(blockchain: Blockchain, hash: Buffer): Promise<Block | undefined> {
+  private async _getHeaderByHash(blockchain: Blockchain, hash: Buffer): Promise<BlockHeader | undefined> {
     try {
-      return blockchain.getBlock(hash)
+      return new Promise(async (resolve) => {
+        let header = (await blockchain.getBlock(hash)).header 
+        resolve(header)
+      })
```

Comment:
Ah, ok. No witchery actually. 😂

Reply:
Do we need to wrap a new Promise here?

Reply:
I actually don't get what this originally did, the original one is an async function but the contents of this function are always executed synchronously; it can thus immediately the `blockchain.getBlock` result, which is a `Promise` itself, but the caller only `await`s on the current function (i.e. `_getBlockByHash`). How did this work in the first place?

Reply:
This is very weird, `catch` could never be reached originally, right? If `getBlock` throws (`NotFoundError`) then the **returned** Promise would throw, but not the actual function itself (i.e. within the try/catch block)